src/tree/ics23_impl.rs

```diff
@@ -254,108 +248,96 @@ mod tests {
-                                                    &new_root_hash.0.to_vec(),
-                                                    b"notexist"
-                                                ));*/
-                                            },
-                                            Middle {
-                                                leftmost_right_proof,
-                                                rightmost_left_proof,
-                                            } => {
-                                                if leftmost_right_proof.root_hash() != new_root_hash {
-                                                    let good_proof = tree.get_with_proof(leftmost_right_proof.leaf().unwrap().key_hash(), 0).unwrap();
-                                                    panic!("root hash mismatch. bad proof: {:?}, good proof: {:?}", leftmost_right_proof, good_proof);
-                                                }
-                                                if rightmost_left_proof.root_hash() != new_root_hash {
-                                                    panic!("root hash mismatch. siblings: {:?}", rightmost_left_proof.siblings());
-                                                }
-
-            /*
-                                                assert!(ics23::verify_non_membership::<HostFunctionsManager>(
-                                                    &commitment_proof,
-                                                    &ics23_spec(),
-                                                    &new_root_hash.0.to_vec(),
-                                                    b"notexist"
-                                                ));*/
-                                            },
-                                        }
-                                    }
-                                }
-
-                    /*
-
-                                assert!(ics23::verify_non_membership::<HostFunctionsManager>(
-                                    &commitment_proof,
-                                    &ics23_spec(),
-                                    &new_root_hash.0.to_vec(),
-                                    b"notexist"
-                                ));*/
-                                /*
-                                assert!(!ics23::verify_non_membership::<HostFunctionsManager>(
-                                    &commitment_proof,
-                                    &ics23_spec(),
-                                    &new_root_hash.0.to_vec(),
-                                    b"key",
-                                )); */
+        #[test]
+        fn test_jmt_ics23_nonexistence(
+            keys: Vec<Vec<u8>>,
+        ) {
+            let db = MockTreeStore::default();
+            let tree = JellyfishMerkleTree::new(&db);
+
+            let mut kvs = Vec::new();
+
+            kvs.push((KeyHash::from(b"key"), Some(b"value1".to_vec())));
```

Comment:
This looks like the cause of the  failing test. The key `b"key"` is inserted into the kvs here (and that is inserted into the tree on line 269). But then on line 337, it's asserted that `b"key"` is not in the resulting trie.